### ISSUE TITLE
loading resources with http

### ISSUE THREAD

**Issue by MenschLink** [crater-invoice/crater#810]
**Describe the bug**
the app tries to load resources with http. I wanted to use this app with a reverse proxy for https in front of the container, but I can't get the app to show in my browser because of mixed content errors
**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/100574825/156196896-43ef082e-73c7-471b-8b55-a9725a3ea098.png)

**Please complete the following information:**
- Crater version: 6.0.4
- PHP version: 7.4.28
- Database type and version: mariadb

**Optional info**
- OS:  [e.g. Ubuntu]
- Browser: [e.g. chrome, safari]

**Comment by mohitpanjwani** [comment]
Hey @MenschLink, 

Can you try to see if this fix works for you? 

https://github.com/crater-invoice/crater/pull/486

Proxies don't work well on Laravel without additional changes.

**Comment by MenschLink** [comment]
it does not... so this is a Laravel issue?

**Comment by mohitpanjwani** [comment]
Which proxy are you using?

**Comment by MenschLink** [comment]
nginx

**Comment by MenschLink** [comment]
so hat I ended up doing now is writing a node script that proxies to the crater app, and  nginx proxies to it:

```nodejs
const http = require("http")
http.createServer((req, res)=>{
        const request = http.request('http://10.13.1.158'+req.url, {method: req.method, headers: req.headers}, (response)=>{
                res.writeHead(response.statusCode, response.headers);
                response.on("error", ()=>{
                        res.statusCode=500;
                        res.end()
                })
                response.on("data", (d)=>{
                        if(d.indexOf(Buffer.from('http:'))>-1){
                                res.write(d.toString().replace(/http:/g, 'https:'))
                        }else{
                                res.write(d)
                        }
                })
                response.on("end", ()=>{
                        res.end()
                })
        })
        request.on("error", ()=>{
                response.statusCode = 500;
                res.end()
        })
        req.pipe(request)
}).listen(3003)
```

**Comment by rm-yakovenko** [comment]
@MenschLink Could you try to add the line `proxy_set_header X-Forwarded-Proto $scheme;` to your nginx config. 

Should be something like that:

```
proxy_set_header Host $host;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
```

**Comment by MenschLink** [comment]
> @MenschLink Could you try to add the line `proxy_set_header X-Forwarded-Proto $scheme;` to your nginx config?
> 
> Should be something like that:
> 
> ```
> proxy_set_header Host $host;
> proxy_set_header X-Real-IP $remote_addr;
> proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
> proxy_set_header X-Forwarded-Proto $scheme;
> ```

this seems to work, thanks!

closing issue

**Comment by lrvick** [comment]
I have the same issue as above. I am using haproxy to terminate TLS, then forward to my craterapp machine.

I tried hardcoding ASSET_URL in .env to https://mydomain.com but this does not seem supported in this laravel setup for some reason.

As a hack, I patched app/Providers/AppServiceProvider.php with:
```
use Illuminate\Support\Facades\URL;

public function boot()
{
       URL::forceScheme('https');
}
```

I would suggest adding a "force https" option in the UI for cases like these, which is what homeassistant and other self-hosted projects do to handle these cases.
